modules/tinymce/src/core/main/ts/delete/BlockMergeBoundary.ts

```diff
@@ -36,14 +38,25 @@ const getBlockPosition = (rootNode: Node, pos: CaretPosition): Optional<BlockPos
 const isDifferentBlocks = (blockBoundary: BlockBoundary): boolean =>
   !Compare.eq(blockBoundary.from.block, blockBoundary.to.block);
 
-const hasSameParent = (blockBoundary: BlockBoundary): boolean =>
-  Traverse.parent(blockBoundary.from.block).bind((parent1) =>
-    Traverse.parent(blockBoundary.to.block).filter((parent2) => Compare.eq(parent1, parent2))
-  ).isSome();
+const getClosestHost = (root: SugarElement<Node>, scope: SugarElement<Node>) => {
```

Comment:
Would it be possible to put a return type for this function

Reply:
I added a type annotation there but ended up fixing a lot more of the types in the file since it was returning a mixed type. Now it is all HTMLElement or SugarElement<HTMLElement> for things.